.github/workflows/kind-e2e.yaml

```diff
@@ -277,6 +277,6 @@ jobs:
         IPS=( $(kubectl get nodes -lkubernetes.io/hostname!=kind-control-plane -ojsonpath='{.items[*].status.addresses[?(@.type=="InternalIP")].address}') )
 
         # Run the tests tagged as e2e on the KinD cluster.
-        go test -race -count=1 -timeout=20m -tags=e2e ${{ matrix.test-suite }} \
+        go test -race -parallel 4 -count=1 -timeout=40m -tags=e2e ${{ matrix.test-suite }} \
```

Comment:
Probably a silly question but I didnt see it called out in the PR description so: what's the reason behind explicitly setting parallel here rather than taking the default?

Reply:
No this was a mistake, it was based on a flawed understanding of how go test works, there's conflicting information online about whether `GOMAXPROCS` or `-parallel` are the controlling factor(s). `go help test` is useless, but `go help testflag` is actually useful and I think we can just let it default to `GOMAXPROCS`